### PR TITLE
Kk/sklearn

### PR DESCRIPTION
**Pull request by kamil-kaczmarek**

### CODE REVIEW

**Comment by HubertJaworski** [comment]
Add publish integration, similar to #456

File: integrations/sklearn/MANIFEST.in [hunk 1]
```diff
@@ -0,0 +1,3 @@
+include versioneer.py
+include neptune/alpha/integrations/sklearn/impl/_version.py
```
Comment (Herudaio) [inline integrations/sklearn/MANIFEST.in:2]: Shouldn't it be new instead of alpha already?

File: neptune/new/integrations/sklearn/__init__.py [hunk 2]
```diff
@@ -0,0 +1,21 @@
...
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+#
+
+try:
+    from neptune_sklearn import create_regressor_summary
```
Comment (Herudaio) [inline neptune/new/integrations/sklearn/__init__.py:18]: Shouldn't we import other methods as well?

File: .github/workflows/neptune-client.yml [hunk 3]
```diff
@@ -45,6 +45,25 @@ jobs:
...
+          python-version: 3.8
+      - name: Build and test
+        run : |
+          cd integrations/sklearn
+          python setup.py sdist
+      - name: Publish package
+        uses: pypa/gh-action-pypi-publish@XXX
+        with:
+          user: __token__
+          password: ${{ secrets.PYPI_API_TOKEN_SKLEARN }}
```
Comment (HubertJaworski) [inline .github/workflows/neptune-client.yml:65]: Follow README.md to create this secret and put it in github project's settings

**Review by HubertJaworski — CHANGES_REQUESTED** [review]

File: .github/workflows/neptune-client.yml [hunk 4]
```diff
@@ -45,6 +45,25 @@ jobs:
...
+      - uses: actions/checkout@v2
+      - uses: actions/setup-python@v2
+        with:
+          python-version: 3.8
+      - name: Build and test
+        run : |
+          cd integrations/sklearn
+          python setup.py sdist
+      - name: Publish package
+        uses: pypa/gh-action-pypi-publish@XXX
```
Comment (HubertJaworski) [inline .github/workflows/neptune-client.yml:62]: `XXX`?

File: integrations/sklearn/setup.py [hunk 5]
```diff
@@ -0,0 +1,37 @@
+import versioneer
```
Comment (aniezurawski) [inline integrations/sklearn/setup.py:1]: Looks like you forgot to modify setup.py in neptune-client.

File: integrations/sklearn/neptune_sklearn/__init__.py [hunk 6]
```diff
@@ -0,0 +1,1190 @@
...
+
+        .. code:: python3
+
+            import neptune.new.integrations.sklearn as npt_utils
+
+            rfc = RandomForestClassifier()
+            rfc.fit(X_train, y_train)
+
+            exp = neptune.init(project='my_workspace/my_project')
+            exp['summary/random_forest'] = npt_utils.create_classifier_summary(rfc, X_train, X_test, y_train, y_test)
```
Comment (Herudaio) [inline integrations/sklearn/neptune_sklearn/__init__.py:173]: Information architecture wise shouldn't it be exp['random_forest/summary']?

File: integrations/sklearn/neptune_sklearn/__init__.py [hunk 7]
```diff
@@ -0,0 +1,1190 @@
...
+
+        .. code:: python3
+
+            import neptune.new.integrations.sklearn as npt_utils
+
+            rfr = RandomForestRegressor()
+            rfr.fit(X_train, y_train)
+
+            exp = neptune.init(project='my_workspace/my_project')
+            exp['summary/random_forest'] = npt_utils.create_regressor_summary(rfr, X_train, X_test, y_train, y_test)
```
Comment (Herudaio) [inline integrations/sklearn/neptune_sklearn/__init__.py:90]: Information architecture wise shouldn't it be exp['random_forest/summary']?

File: integrations/sklearn/neptune_sklearn/__init__.py [hunk 8]
```diff
@@ -0,0 +1,1190 @@
...
+    Examples:
+        .. code:: python3
+
+            import neptune.new.integrations.sklearn as npt_utils
+
+            km = KMeans(n_init=11, max_iter=270)
+            X, y = make_blobs(n_samples=579, n_features=17, centers=7, random_state=28743)
+
+            exp = neptune.init(project='my_workspace/my_project')
+            exp['summary/kmeans'] = npt_utils.create_kmeans_summary(km, X)
```
Comment (Herudaio) [inline integrations/sklearn/neptune_sklearn/__init__.py:237]: Information architecture wise shouldn't it be exp['kmeans/summary']?

File: .github/workflows/neptune-client.yml (hunk 3 above)
Comment (kamil-kaczmarek) [reply .github/workflows/neptune-client.yml:65]: @HubertJaworski - I cannot do it, as I do not have access to pypi account.

File: .github/workflows/neptune-client.yml (hunk 4 above)
Comment (kamil-kaczmarek) [reply .github/workflows/neptune-client.yml]: done

File: integrations/sklearn/neptune_sklearn/__init__.py [hunk 9]
```diff
@@ -0,0 +1,1190 @@
...
+    Examples:
+        .. code:: python3
+
+            import neptune.new.integrations.sklearn as npt_utils
+
+            rfc = RandomForestClassifier()
+            rfc.fit(X_train, y_train)
+
+            exp = neptune.init(project='my_workspace/my_project')
+            exp['report'] = npt_utils.create_confusion_matrix_chart(rfc, X_train, X_test, y_train, y_test)
```
Comment (Herudaio) [inline integrations/sklearn/neptune_sklearn/__init__.py:860]: `report` -> `visuals/confusion_matrix`?

File: integrations/sklearn/neptune_sklearn/__init__.py [hunk 10]
```diff
@@ -0,0 +1,1190 @@
...
+    Examples:
+        .. code:: python3
+
+            import neptune.new.integrations.sklearn as npt_utils
+
+            rfc = RandomForestClassifier()
+            rfc.fit(X_train, y_train)
+
+            exp = neptune.init(project='my_workspace/my_project')
+            exp['report'] = npt_utils.create_roc_auc_chart(rfc, X_train, X_test, y_train, y_test)
```
Comment (Herudaio) [inline integrations/sklearn/neptune_sklearn/__init__.py:912]: `report` -> `visuals/roc_auc_chart`?

File: integrations/sklearn/neptune_sklearn/__init__.py [hunk 11]
```diff
@@ -0,0 +1,1190 @@
...
+    Examples:
+        .. code:: python3
+
+            import neptune.new.integrations.sklearn as npt_utils
+
+            rfc = RandomForestClassifier()
+            rfc.fit(X_train, y_train)
+
+            exp = neptune.init(project='my_workspace/my_project')
+            exp['report'] = npt_utils.create_precision_recall_chart(rfc, X_test, y_test)
```
Comment (Herudaio) [inline integrations/sklearn/neptune_sklearn/__init__.py:962]: `report` -> `visuals/rprecision_recall_chart`?

File: integrations/sklearn/neptune_sklearn/__init__.py [hunk 12]
```diff
@@ -0,0 +1,1190 @@
...
+    Examples:
+        .. code:: python3
+
+            import neptune.new.integrations.sklearn as npt_utils
+
+            rfc = RandomForestClassifier()
+            rfc.fit(X_train, y_train)
+
+            exp = neptune.init(project='my_workspace/my_project')
+            exp['report'] = npt_utils.create_class_prediction_error_chart(rfc, X_train, X_test, y_train, y_test)
```
Comment (Herudaio) [inline integrations/sklearn/neptune_sklearn/__init__.py:1019]: `report` -> `visuals/class_prediction_error_chart`?

File: integrations/sklearn/setup.py (hunk 5 above)
Comment (HubertJaworski) [reply integrations/sklearn/setup.py:1]: everything needed to be done is described in `integgrations/README.md`. We can create the pypi token after merging this commit, but there's still some things missing from that instructions

File: .github/workflows/neptune-client.yml (hunk 3 above)
Comment (HubertJaworski) [reply .github/workflows/neptune-client.yml:65]: send me a package in version 0.0.0 and I set up the secret

File: integrations/sklearn/MANIFEST.in (hunk 1 above)
Comment (kamil-kaczmarek) [reply integrations/sklearn/MANIFEST.in:2]: I followed template provided by @HubertJaworski. Should I correct it @HubertJaworski ?

File: neptune/new/integrations/sklearn/__init__.py (hunk 2 above)
Comment (kamil-kaczmarek) [reply neptune/new/integrations/sklearn/__init__.py:18]: Good question. @HubertJaworski ?

File: integrations/sklearn/neptune_sklearn/__init__.py (hunk 6 above)
Comment (kamil-kaczmarek) [reply integrations/sklearn/neptune_sklearn/__init__.py:173]: corrected.

File: integrations/sklearn/neptune_sklearn/__init__.py (hunk 7 above)
Comment (kamil-kaczmarek) [reply integrations/sklearn/neptune_sklearn/__init__.py:90]: corrected.

File: integrations/sklearn/neptune_sklearn/__init__.py (hunk 8 above)
Comment (kamil-kaczmarek) [reply integrations/sklearn/neptune_sklearn/__init__.py:237]: corrected.

File: integrations/sklearn/neptune_sklearn/__init__.py (hunk 9 above)
Comment (kamil-kaczmarek) [reply integrations/sklearn/neptune_sklearn/__init__.py:860]: corrected.

File: integrations/sklearn/neptune_sklearn/__init__.py (hunk 11 above)
Comment (kamil-kaczmarek) [reply integrations/sklearn/neptune_sklearn/__init__.py:962]: corrected.

File: integrations/sklearn/neptune_sklearn/__init__.py (hunk 12 above)
Comment (kamil-kaczmarek) [reply integrations/sklearn/neptune_sklearn/__init__.py:1019]: corrected.

File: integrations/sklearn/neptune_sklearn/__init__.py (hunk 10 above)
Comment (kamil-kaczmarek) [reply integrations/sklearn/neptune_sklearn/__init__.py:913]: corrected.

File: integrations/sklearn/setup.py (hunk 5 above)
Comment (kamil-kaczmarek) [reply integrations/sklearn/setup.py:1]: added sklearn to setup.py

File: .github/workflows/neptune-client.yml (hunk 3 above)
Comment (kamil-kaczmarek) [reply .github/workflows/neptune-client.yml:65]: done

File: integrations/sklearn/MANIFEST.in (hunk 1 above)
Comment (kamil-kaczmarek) [reply integrations/sklearn/MANIFEST.in:2]: fixed

File: neptune/new/integrations/sklearn/__init__.py (hunk 2 above)
Comment (kamil-kaczmarek) [reply neptune/new/integrations/sklearn/__init__.py:18]: fixed

File: neptune/new/integrations/sklearn/__init__.py [hunk 13]
```diff
@@ -0,0 +1,27 @@
...
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+#
+
+try:
+    from neptune_sklearn import create_regressor_summary, create_classifier_summary, create_kmeans_summary, \
```
Comment (HubertJaworski) [inline neptune/new/integrations/sklearn/__init__.py:18]: Currently we prefer import style:
```
from module import(
   function_1,
   function_2,
   function_3
)
```

**Review by HubertJaworski — COMMENTED** [review]
Changes look good from the point of view of integration/deployment. One comment on integration in core lib.

I'd like someone else to take a look at the actual integration, maybe @jakubczakon ?

File: neptune/new/integrations/sklearn/__init__.py (hunk 13 above)
Comment (kamil-kaczmarek) [reply neptune/new/integrations/sklearn/__init__.py]: done

**Comment by kamil-kaczmarek** [comment]
@HubertJaworski commited code formatting changes.

For the actual integration there is @Herudaio who looks at it as well.

If looks good from your point of view pls accept.

**Review by Herudaio — APPROVED** [review]

**Review by HubertJaworski — APPROVED** [review]

**Pull request merged**